### ISSUE TITLE
bug: some select-text keyboard actions and keyboard navigation does not work inside edit-code popup

### ISSUE THREAD

**Issue by olejorgenb** [sourcegraph/jetbrains#2245]
### IDE Information

PyCharm 2024.1.4 (Professional Edition)
Build #PY-241.18034.82, built on June 24, 2024
Licensed to ******* / Ole Jørgen Brønner
Runtime version: 17.0.11+1-b1207.24 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 6.8.0-40-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 2560M
Cores: 4
Registry:
  python.debug.low.impact.monitoring.api=False
Non-Bundled Plugins:
  com.berzanmikaili.intellij.monokai_pro (2.0.2c)
  MultiHighlight (3.2.0)
  org.intellij.plugins.hcl (241.17011.119)
  Shifter (1.9.6)
  com.ultrahob.zerolength.plugin (0.95)
  org.jetbrains.action-tracker (0.3.3)
  com.intellij.ideolog (222.3.2.0)
  com.tylerthrailkill.intellij.solarized (3.0.0)
  org.jeto.taskfocus (1.2)
  com.arcticicestudio.nord.jetbrains (0.13.0)
  GrepConsole (13.1.0-IJ2023.3)
  uk.co.ben-gibson.remote.repository.mapper (4.5.1)
  com.sourcegraph.jetbrains (7.0.0)
  Activity Tracker (0.1.12 beta)
  com.insyncwithfoo.pyrightls (1.1.0)
Current Desktop: ubuntu:GNOME

### Bug Description

Write a few lines inside the popup and try to select them with `Shift-ArrowUp` - does not work.

![image](https://github.com/user-attachments/assets/e7944ef1-a0e2-44d2-b272-a49994fb1504)

Arrow navigation in the lines are broken for me as well. If I press `Down` in the above scenario the cursor jumps to the empty line between "line 2" and "line 3"...

**Comment by olejorgenb** [comment]
~~Works in PyCharm 2024.2.3 (Professional Edition) with Cody 7.0.13.~~

Only tested with the newstyle UI (ie. have not tested with the old-style UI plugin - I've resigned to the new pycharm UI)

EDIT: spoke to soon - something very funky is still going on :/
